hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContractRevision.php

```diff
@@ -262,6 +262,17 @@ protected static function buildFullDetailsQuery($entities, $revision) {
 
       $query['from'][] = $table;
       $query['where'][] = "{$table}.jobcontract_revision_id = " . $revision["{$entity}_revision_id"];
+
+      if ($entity == 'pension') {
+        $query['select'][] = "pension_ov.label as {$entity}__pension_type_label";
+
+        $query['from'][] = 'civicrm_option_group pension_og';
+        $query['from'][] = 'civicrm_option_value pension_ov';
+
+        $query['where'][] = "pension_og.name = 'hrjc_pension_type'";
+        $query['where'][] = 'pension_ov.option_group_id = pension_og.id';
+        $query['where'][] = "pension_ov.value = {$table}.pension_type";
+      }
```

Comment:
sorry I missed this .. @MiyaNoctem  , can u please not just fixing this tests .. but also write some tests for fullDetails BAO method .. it would make more sense to test the BAO method directly